atlasdb-config/src/test/java/com/palantir/atlasdb/factory/TransactionManagersTest.java

```diff
@@ -714,6 +722,88 @@ public void kvsDoesNotRecordSweepStatsIfSweepQueueWritesAndTargetedSweepEnabled(
+
+        AtlasDbConfig installConfig = ImmutableAtlasDbConfig.builder()
+                .keyValueService(new InMemoryAtlasDbConfig())
+                .targetedSweep(ImmutableTargetedSweepInstallConfig.builder().build())
+                .timestampCache(expectedTimestampCache)
+                .build();
+
+        TimestampCache timestampCache = constructTimestampCache(installConfig, Optional.empty());
+
+        assertThat(timestampCache).isSameAs(expectedTimestampCache);
+    }
+
+    @Test
+    public void usingInMemoryTimestampCache() {
+        AtlasDbConfig installConfig = ImmutableAtlasDbConfig.builder()
+                .keyValueService(new InMemoryAtlasDbConfig())
+                .targetedSweep(ImmutableTargetedSweepInstallConfig.builder().build())
+                .build();
+
+        TimestampCache timestampCache = constructTimestampCache(installConfig, Optional.empty());
+
+        assertThat(timestampCache).isInstanceOf(DefaultTimestampCache.class);
+    }
+
+    @Test
+    public void usingPersistentStorage() throws IOException {
+        File storageFolder = temporaryFolder.newFolder();
+        String storagePath = Files.currentFolder().toPath()
+                .relativize(storageFolder.toPath())
+                .toString();
+
+        AtlasDbConfig installConfig = ImmutableAtlasDbConfig.builder()
+                .keyValueService(new InMemoryAtlasDbConfig())
+                .targetedSweep(ImmutableTargetedSweepInstallConfig.builder().build())
+                .persistentStorage(
+                        ImmutableRocksDbPersistentStorageConfig.builder()
+                                .storagePath(storagePath)
+                                .build())
+                .build();
+
+        Optional<PersistentTimestampStore> persistentTimestampStore =
+                TransactionManagers.constructPersistentTimestampStore(
+                        installConfig,
+                        new DefaultPersistentStorageFactory(),
+                        new LinkedList<>());
+
+        TimestampCache timestampCache = constructTimestampCache(installConfig, persistentTimestampStore);
+
+        assertThat(timestampCache).isInstanceOf(OffHeapTimestampCache.class);
+    }
+
+    @Test
+    public void persistentTimestampStoreNotConstructed() {
+        AtlasDbConfig installConfig = ImmutableAtlasDbConfig.builder()
+                .keyValueService(new InMemoryAtlasDbConfig())
+                .targetedSweep(ImmutableTargetedSweepInstallConfig.builder().build())
+                .build();
+
+        Optional<PersistentTimestampStore> persistentTimestampStore =
+                TransactionManagers.constructPersistentTimestampStore(
+                        installConfig,
+                        new DefaultPersistentStorageFactory(),
+                        new LinkedList<>());
+
+        assertThat(persistentTimestampStore)
+                .isEmpty();
+    }
+
```

Comment:
There's probably something around `persistentTimestampStoreNotConstructedEvenIfConfiguredIfExplicitlyProvided`, though I'm okay with passing on that if it's really hard to test.

Reply:
Will be resolved in a following PR.

Reply:
After discussing offline we decided that if the config is provided the persistent storage is initialized so this is not useful anymore.